pkg/distributor/query.go

```diff
@@ -184,6 +211,81 @@ func (d *Distributor) queryIngesters(ctx context.Context, replicationSet ring.Re
 	return result, nil
 }
 
+// mergeExemplarSets merges and dedupes two sets of already sorted sample pairs.
+// Defined here instead of pkg/util to avoid a import cycle.
```

Comment:
Might be useful to comment here that all exemplars are expected to be for the same series.